sky/jobs/utils.py

```diff
@@ -732,7 +732,12 @@ def should_keep_logging(status: managed_job_state.ManagedJobStatus) -> bool:
                         if log_lib.LOG_FILE_START_STREAMING_AT in line:
                             start_streaming = True
                         if start_streaming:
-                            print(line, end='', flush=True)
+                            print(line, end='' if line.endswith('\r') else '\n', flush=True)
```

Comment:
We need to be careful with this. If the line ends with `\r\n`, this may cause unexpeted logging.